### PR TITLE
Spinnaker 3.0.0.118 upgrade

### PR DESCRIPTION
**Pull request by ds58**
I upgraded the spinnaker preset to the latest Spinnaker SDK version `3.0.0.118`

Had to change a few things in the sample code, nothing major.

I added a few skips for some enum symbols I wasn't sure how to deal with:
```
enum TLPayloadType
{
    ...
    SPINNAKER_TLPAYLOAD_TYPE_CUSTOM_ID = 1000, /* Starting value for GenTL Producer custom IDs*/
    SPINNAKER_TLPAYLOAD_TYPE_LOSSLESS_COMPRESSED = SPINNAKER_TLPAYLOAD_TYPE_CUSTOM_ID + 1,
    SPINNAKER_TLPAYLOAD_TYPE_LOSSY_COMPRESSED = SPINNAKER_TLPAYLOAD_TYPE_CUSTOM_ID + 2,
    SPINNAKER_TLPAYLOAD_TYPE_JPEG_LOSSLESS_COMPRESSED = SPINNAKER_TLPAYLOAD_TYPE_CUSTOM_ID + 3
```
In Java, it maps to
```
public enum spinTLPayloadType {
    ...
    SPINNAKER_TLPAYLOAD_TYPE_CUSTOM_ID(1000), /* Starting value for GenTL Producer custom IDs*/
    SPINNAKER_TLPAYLOAD_TYPE_LOSSLESS_COMPRESSED(SPINNAKER_TLPAYLOAD_TYPE_CUSTOM_ID + 1),
    SPINNAKER_TLPAYLOAD_TYPE_LOSSY_COMPRESSED(SPINNAKER_TLPAYLOAD_TYPE_CUSTOM_ID + 2),
    SPINNAKER_TLPAYLOAD_TYPE_JPEG_LOSSLESS_COMPRESSED(SPINNAKER_TLPAYLOAD_TYPE_CUSTOM_ID + 3);
```
I couldn't find any examples on how to deal with "self referencing enums." Any thoughts?

### CODE REVIEW

**Comment by saudet** [comment]
> I couldn't find any examples on how to deal with "self referencing enums." Any thoughts?

That should be working. It's a bug in JavaCPP, I guess, but are those enums important?

**Comment by ds58** [comment]
Those enums aren't really important from what I can tell

**Comment by calvertdw** [comment]
It looks like an outdated actions.yml was used for the build, pulling the old Spinnaker downloads. :thinking:

**Comment by saudet** [comment]
To test that, we would need to replace *temporarily* `bytedeco/javacpp-presets/.github/actions/deploy-[...]@actions` with `ihmcrobotics/javacpp-presets/.github/actions/deploy-[...]@spinnaker-upgrade` in https://github.com/ihmcrobotics/javacpp-presets/blob/spinnaker-upgrade/.github/workflows/spinnaker.yml

File: spinnaker/README.md [hunk 1]
```diff
@@ -86,571 +86,578 @@ import static org.bytedeco.spinnaker.global.Spinnaker_C.*;
...
-            BytePointer featureValue = new BytePointer(MAX_BUFF_LEN);
-            SizeTPointer lenFeatureValue = new SizeTPointer(1);
-            lenFeatureValue.put(MAX_BUFF_LEN);
-            err = spinNodeToString(hFeatureNode, featureValue, lenFeatureValue);
-            if (printOnError(err, "spinNodeToString")) {
-                featureValue.putString("Unknown value");
+   private final static int MAX_BUFF_LEN = 256;
+
+   private static String findErrorNameByValue(int value) {
+      for (spinError v : spinError.values()) {
```
Comment (saudet) [inline spinnaker/README.md:92]: Please undo the unnecessary formatting changes in the README.md file.

File: spinnaker/src/main/java/org/bytedeco/spinnaker/presets/Spinnaker_C.java [hunk 2]
```diff
@@ -95,6 +95,9 @@ public void map(InfoMap infoMap) {
                // Skip to avoid linker errors, somehow Spinnaker SDK does not export those functions,
                // To avoid errors like: jniSpinnaker_C.obj : error LNK2001: unresolved external symbol spinCameraForceIP
                .put(new Info("spinCameraForceIP", "spinRegisterSetEx", "spinSystemSendActionCommand", "spinInterfaceSendActionCommand").skip())
+               .put(new Info("SPINNAKER_TLPAYLOAD_TYPE_LOSSLESS_COMPRESSED").skip())
+               .put(new Info("SPINNAKER_TLPAYLOAD_TYPE_LOSSY_COMPRESSED").skip())
+               .put(new Info("SPINNAKER_TLPAYLOAD_TYPE_JPEG_LOSSLESS_COMPRESSED").skip())
```
Comment (saudet) [inline spinnaker/src/main/java/org/bytedeco/spinnaker/presets/Spinnaker_C.java:100]: Let's put all that in the same entry, something like this:
```java
               .put(new Info("spinCameraForceIP", "spinRegisterSetEx", "spinSystemSendActionCommand", "spinInterfaceSendActionCommand",
                             "SPINNAKER_TLPAYLOAD_TYPE_LOSSLESS_COMPRESSED",
                             "SPINNAKER_TLPAYLOAD_TYPE_LOSSY_COMPRESSED",
                             "SPINNAKER_TLPAYLOAD_TYPE_JPEG_LOSSLESS_COMPRESSED").skip())
```

Comment (ds58) [reply spinnaker/src/main/java/org/bytedeco/spinnaker/presets/Spinnaker_C.java]: Whoops, I've fixed that.

**Comment by saudet** [comment]
Could we get this tested as per https://github.com/bytedeco/javacpp-presets/pull/1313#issuecomment-1397886832?

File: .github/actions/deploy-centos/action.yml [hunk 3]
```diff
@@ -145,9 +145,9 @@ runs:
 
         if [[ "$CI_DEPLOY_PLATFORM" == "linux-x86_64" ]] && [[ "$CI_DEPLOY_MODULE" == "spinnaker" ]]; then
           echo Installing Spinnaker amd64
-          python3 -m gdown.cli https://drive.google.com/uc?id=1zr9k4x-NI3OkVMLCccSBY4VtR-16iAQG
-          tar -xzvf spinnaker-2.4.0.143-Ubuntu18.04-amd64-pkg.tar.gz
-          find -name *.deb | while read f; do ar -xv $f; tar --totals -xvf data.tar.xz -C /; done;
+          python3 -m gdown.cli https://drive.google.com/uc?id=1ILDEZk0SuOoh4_cJExpZJCPXCqfxUkAi
+          tar -xzvf spinnaker-3.0.0.118-amd64-pkg.tar.gz
+          find -name *.deb | while read f; do ar -xv $f; tar -I zstd --totals -xvf data.tar.zst -C /; done;
```
Comment (saudet) [inline .github/actions/deploy-centos/action.yml:150]: The files available from https://flir.app.boxcn.net/v/SpinnakerSDK/folder/133154514756 differ, why?

Comment (ds58) [reply .github/actions/deploy-centos/action.yml:150]: That's very odd. I didn't change anything before uploading strait to Drive. I'll upload fresh and change the file ID.

Comment (saudet) [reply .github/actions/deploy-centos/action.yml:150]: It looks like what changed is they reverted switching to zstd from xz, so let's also remove the changes related to that:
```
x - debian-binary
x - control.tar.xz
x - data.tar.xz
tar (child): data.tar.zst: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
```

Comment (ds58) [reply .github/actions/deploy-centos/action.yml]: Ah, yep that would make sense. Haha.

**Comment by saudet** [comment]
Ok, this looks good to merge. Please revert https://github.com/bytedeco/javacpp-presets/pull/1313/commits/2f0759d24f73e3742914fe8335390bc5e5f5581b and let's get this in!
